src/main/java/uk/gov/companieshouse/api/strikeoffobjections/controller/ObjectionController.java

```diff
@@ -195,6 +199,45 @@ public ResponseEntity patchObjection(
         }
     }
 
+    @GetMapping("/{objectionId}")
+    public ResponseEntity<ChResponseBody<ObjectionResponseDTO>> getObjection(
+            @PathVariable("companyNumber") String companyNumber,
+            @PathVariable("objectionId") String objectionId,
+            @RequestHeader(value = ERIC_REQUEST_ID_HEADER) String requestId
+    ) {
+        Map<String, Object> logMap = new HashMap<>();
+        logMap.put(LOG_COMPANY_NUMBER_KEY, companyNumber);
+        logMap.put(LOG_OBJECTION_ID_KEY, objectionId);
+
+        apiLogger.infoContext(
+                requestId,
+                "GET /{objectionId} request received",
+                logMap
+        );
+
+        try {
+            Objection objection = objectionService.getObjection(requestId, objectionId);
+            ObjectionResponseDTO responseDTO =
+                    objectionMapper.objectionEntityToObjectionResponseDTO(objection);
+            return responseEntityFactory.createResponse(ServiceResult.found(responseDTO));
+        } catch (ObjectionNotFoundException e) {
+            apiLogger.errorContext(
+                    requestId,
+                    OBJECTION_NOT_FOUND,
+                    e,
+                    logMap
+            );
+
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
```

Comment:
I would add another 'catch all' catch(Exception e) to handle anything other than not found exceptions. This would log the error and return a 500 error back to the user

Reply:
it would go after the not found catch block

Reply:
Agree. We should add this to all the other controller end-point methods too, as it seems to be missing on most of them. Another sub-task and PR perhaps...

Reply:
Added another catch block for the 500 error. We can maybe add this check into the other methods in a new story

Reply:
Could be worth creating a global restControllerAdvice exception handler, and just let the controllers throw the exceptions and the global handler will catch and handle them all, so we don't duplicate similar code throughout the controller.